### PR TITLE
Fix for empty column on replace and out of memory catching for join and tab

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Added a Panic.catch to catch heap memory error in joins and cross_tab.
- Adjusted column replace so type is correct.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso [hunk 1]
```diff
@@ -380,3 +381,20 @@ type Dry_Run_Operation
...
+    ## PRIVATE
+       Convert the Out_Of_Memory to a human-readable format.
+    to_display_text : Text
+    to_display_text self = "Enso ran out of memory performing the "+self.operation+" operation."
+
+    ## PRIVATE
+       ADVANCED
+       Catches possible `OutOfMemoryError` and throws an `Out_Of_Memory` error.
+    handle_java_exception operation ~function =
+        Panic.catch OutOfMemoryError handler=(_-> Error.throw (Out_Of_Memory.Error operation)) <|
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso:399]: Isn't it generally frowned-upon to try to catch OOM?

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso (hunk 1 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso:399]: Catching in a few limited places where the user can easily create a very large table and cause the program to crash.

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso:399]: Shouldn't we rather try to check if the allocation is safe _before_ performing it? (i.e. by checking available runtime memory)

Indeed it seems rather unsafe to catch OOME: https://stackoverflow.com/a/8729011 - the JVM can be left in an inconsistent state

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso:399]: Happy to add a ticket to revisit all the allocations within Table operations but that is a massive task and we should probably do as part of the restructuring work.

**Pull request merged**